### PR TITLE
Return field value from useSpy

### PR DESCRIPTION
**Pull request by arempe93**
Changes the return value of `useSpy` from `void` to an object containing the field's value

### Motivation

I had a use case like this (simplified):

```js
const MaxLengthSpy = ({ fieldId, maxLength }) => {
  useSpy(fieldId, (value: string, { setFieldError }) => {
    if (value.length > maxLength) {
      setFieldError(fieldId, 'too long')
    }
  })

  return (
    <Flex justify='flex-end'>
      <Text color='grey50' size={0.75}>
        {/* whats the current length? */} / {maxLength}
      </Text>
    </Flex>
  )
}
```

Seems logical that `useSpy` could return the value, since it would need it to invoke the callback

### v5 / Large minor patch ideas

Getting a `FormHookContext` update only when a specific field changes is a great concept. The alternative of `useFormConnect` is inefficient and clumsy. Unfortunately, in its current state, `useSpy` is very limiting.

It would be great if `useSpy` could just return the context value, instead of passing it to a callback. This allows `useSpy` to truly be a more efficient `useFormConnect` instead.

As far as I can tell, there would be no drawbacks to this approach (besides API changes). The hook is connected to the context anyways, and current behavior could be achieved simply by using built-in React hooks.

If you prefer this idea, I can replace this PR with one that achieves this with new tests, etc. Could be achieved in v4 with a deprecation notice for `useSpy`s second argument

### CODE REVIEW

File: src/useSpy.ts [hunk 1]
```diff
@@ -3,13 +3,20 @@ import { get } from './helpers/operations';
 import { FormHookContext } from './types';
 import { useContextEmitter } from './useContextEmitter';
 
-const useSpy = (fieldId: string, cb: (newValue: any, ctx: FormHookContext) => void) => {
+const useSpy = (
+  fieldId: string,
+  cb: (newValue: any, ctx: FormHookContext) => void
+) => {
```
Comment (arempe93) [inline src/useSpy.ts]: prettier did this

**Comment by JoviDeCroock** [comment]
Hey,

I appreciate the idea and wonder if this would really need a major, we can just return the value and other metadata and still support the callback as far as I know.

Thank you so much for taking time out of your day to do this thorough writeup and PR, I’ll give it some more thought as to how we can make useSpy even better. :+1:

**Comment by arempe93** [comment]
Hey thanks for taking the time to think over `useSpy` changes. I updated the PR to change the return value to what the callback receives, which I thought was logical. And also updating the typings for `useSpy`, which were a bit stale.

I'm just now seeing your edit tho, which gives me a bit of pause on returning context.

I really like your suggestion of returning `FieldInformation` like in `useField`, if that's what you mean by fieldMetadata. It could even replace `useError` if the callback were optional -- or if `useError` could leverage the same callback functionality.

eg:
```js
// instead of useError('myField')
const { error } = useSpy('myField', (ctx) => /* callback if I want? */)
```

---

Answering your question about more experiences with the library: the entire reason I'm down this path is because you can't do validations (ie `setFieldError`) within a `useField` hooked component. I think that would be a good addition to `FieldOperations` imo.

**Comment by JoviDeCroock** [comment]
Do you mean arbirarly setting a fieldError on ant field or one bound to just that field like the onChange?

**Comment by JoviDeCroock** [comment]
@all-contributors please add arempe93 for code

**Pull request merged**